transports/p2p/addrmgr/addrmanager.go

```diff
@@ -211,6 +227,48 @@ func (a *AddrManager) updateAddress(netAddr, srcAddr *wire.NetAddress) {
 	a.Log.Trace().Msgf("Added new address %s for a total of %d addresses", addr, a.nTried+a.nNew)
 }
 
+// BanAddress creates a ban for defaultBanTime for a peer that we cannot connect to.
+func (a *AddrManager) BanAddress(addr string) {
```

Comment:
Does it need to be public?

Reply:
We pass this function to connManager in `tranports/p2p/server.go` line 1441, using the server's addrManager, so I think it needs to be exported publicly. Also for testing purposes.